From 4510afffc493f03bffb16c61984ba924420d6c65 Mon Sep 17 00:00:00 2001
From: Pogobanane <aenderboy@gmx.de>
Date: Thu, 3 Jun 2021 15:59:35 +0000
Subject: [PATCH] unset cputemplate

---
 config/config.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/config.go b/config/config.go
index ef45032..c8993e8 100644
--- a/config/config.go
+++ b/config/config.go
@@ -87,7 +87,7 @@ func LoadConfig(path string) (*Config, error) {
 		KernelArgs:      defaultKernelArgs,
 		KernelImagePath: defaultKernelPath,
 		RootDrive:       defaultRootfsPath,
-		CPUTemplate:     string(defaultCPUTemplate),
+		CPUTemplate:     "",
 		ShimBaseDir:     defaultShimBaseDir,
 		JailerConfig: JailerConfig{
 			RuncConfigPath: runcConfigPath,
-- 
2.31.1

